### ISSUE TITLE
disableLocationAuthorizationAlert it's not working as expected

### ISSUE THREAD

**Issue by giovanegomes** [transistorsoft/react-native-background-geolocation#1791]
## Your Environment
* Plugin version: 4.12.1
* Platform: iOS or Android
* OS version: IOS 16.4 - Android 13, 11
* Device manufacturer / model: Xiomi redmi 11, Samsung galaxy M62 
* React Native version (`react-native -v`): 10.2.4
* Plugin config
```javascript <!-- Syntax-highlighting:  paste your code below -->
{
  desiredAccuracy: BackgroundGeolocation.DESIRED_ACCURACY_HIGH,
  distanceFilter: 5,
  preventSuspend: false,
  stopTimeout: 5,
  stationaryRadius: 10,
  isMoving: true,
  logLevel: BackgroundGeolocation.LOG_LEVEL_VERBOSE,
  notification: { text: 'Caminhamento ativado.' },
  disableLocationAuthorizationAlert: true,
  locationAuthorizationAlert: {
    titleWhenNotEnabled:
      'O serviço de localização em segundo plano está desabilitado.',
    titleWhenOff: 'Os serviços de localização estão OFF.',
    instructions:
      'Para utilizar os serviços de localização em segundo plano é necessário habilitar "Sempre" nas Configurações - Serviços de Localização',
    cancelButton: 'Cancelar',
    settingsButton: 'Configurações',
  },
  backgroundPermissionRationale: {
    title:
      'Permitir que o {applicationName} acesse a localização do dispositivo mesmo quando ele estiver fechado ou não estiver sendo utilizado?',
    message:
      'O aplicativo utiliza a localização para obter os pontos do caminhamento.',
    positiveAction: 'Alterar para "{backgroundPermissionOptionLabel}"',
    negativeAction: 'Cancelar',
  },
}
```

## Expected Behavior

When setting the config `disableLocationAuthorizationAlert` to `true`, we expected to avoid the alert asking to change the permission when executing `BackgroundGeolocation.start()`.

## Actual Behavior

Plugin is asking to change the permission twice, first when executing `BackgroundGeolocation.requestPermission()` and then when executing `BackgroundGeolocation.start()`. We tried to disable this alert by setting the config `disableLocationAuthorizationAlert` to `true`, but it's still showing it.

## Steps to Reproduce

```js
const state = await BackgroundGeolocation.ready(GEOLOCATION_CONFIG);

if (state.enabled) return;

try {
    await BackgroundGeolocation.requestPermission();
} catch (error) {
    console.error('Request permission', error);
}

await this.requestPermissions();
await BackgroundGeolocation.start();
```

## Context

We are trying to manually manage the location permissions

## Debug logs

<details><summary>Logs</summary>

``` <!-- Syntax highlighting:  DO NOT REMOVE -->
08-23 11:16:40.389 25484 25569 D TSLocationManager: [c.t.l.logger.LoggerFacade$a a] PID: 2
08-23 11:16:40.389 25484 25570 D TSLocationManager: [c.t.l.l.TSSQLiteAppender$c run] 
08-23 11:16:40.389 25484 25570 D TSLocationManager:   ℹ️  Cleared logs older than 72 hours
08-23 11:16:40.390 25484 25569 I TSLocationManager: [c.t.l.logger.LoggerFacade$a a] 
08-23 11:16:40.390 25484 25569 I TSLocationManager: ╔═════════════════════════════════════════════
08-23 11:16:40.390 25484 25569 I TSLocationManager: ║ TSLocationManager version: 3.4.0 (420)
08-23 11:16:40.390 25484 25569 I TSLocationManager: ╠═════════════════════════════════════════════
08-23 11:16:40.390 25484 25569 I TSLocationManager: ╟─ Xiaomi 2201117TG @ 13 (react-native)
08-23 11:16:40.390 25484 25569 I TSLocationManager: {
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "activityRecognitionInterval": 10000,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "allowIdenticalLocations": false,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "authorization": {},
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "autoSync": true,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "autoSyncThreshold": 0,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "backgroundPermissionRationale": {
08-23 11:16:40.390 25484 25569 I TSLocationManager:     "title": "Permitir que o {applicationName} acesse a localização do dispositivo mesmo quando ele estiver fechado ou não estiver sendo utilizado?",
08-23 11:16:40.390 25484 25569 I TSLocationManager:     "message": "O aplicativo utiliza a localização para obter os pontos do caminhamento.",
08-23 11:16:40.390 25484 25569 I TSLocationManager:     "positiveAction": "Alterar para \"{backgroundPermissionOptionLabel}\"",
08-23 11:16:40.390 25484 25569 I TSLocationManager:     "negativeAction": "Cancelar"
08-23 11:16:40.390 25484 25569 I TSLocationManager:   },
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "batchSync": false,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "configUrl": "",
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "debug": false,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "deferTime": 0,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "desiredAccuracy": -1,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "desiredOdometerAccuracy": 100,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "disableAutoSyncOnCellular": false,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "disableElasticity": false,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "disableLocationAuthorizationAlert": true,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "disableMotionActivityUpdates": false,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "disableProviderChangeRecord": false,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "disableStopDetection": false,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "distanceFilter": 5,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "elasticityMultiplier": 1,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "enableHeadless": false,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "enableTimestampMeta": false,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "extras": {},
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "fastestLocationUpdateInterval": -1,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "foregroundService": true,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "geofenceInitialTriggerEntry": true,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "geofenceModeHighAccuracy": false,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "geofenceProximityRadius": 1000,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "geofenceTemplate": "",
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "headers": {},
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "headlessJobService": "com.transistorsoft.rnbackgroundgeolocation.HeadlessTask",
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "heartbeatInterval": -1,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "httpRootProperty": "location",
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "httpTimeout": 60000,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "isMoving": true,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "locationAuthorizationRequest": "Always",
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "locationTemplate": "",
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "locationTimeout": 60,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "locationUpdateInterval": 1000,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "locationsOrderDirection": "ASC",
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "logLevel": 5,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "logMaxDays": 3,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "maxBatchSize": -1,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "maxDaysToPersist": 1,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "maxRecordsToPersist": -1,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "method": "POST",
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "minimumActivityRecognitionConfidence": 75,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "motionTriggerDelay": 0,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "notification": {
08-23 11:16:40.390 25484 25569 I TSLocationManager:     "layout": "",
08-23 11:16:40.390 25484 25569 I TSLocationManager:     "title": "",
08-23 11:16:40.390 25484 25569 I TSLocationManager:     "text": "Caminhamento ativado.",
08-23 11:16:40.390 25484 25569 I TSLocationManager:     "color": "",
08-23 11:16:40.390 25484 25569 I TSLocationManager:     "channelName": "TSLocationManager",
08-23 11:16:40.390 25484 25569 I TSLocationManager:     "channelId": "",
08-23 11:16:40.390 25484 25569 I TSLocationManager:     "smallIcon": "",
08-23 11:16:40.390 25484 25569 I TSLocationManager:     "largeIcon": "",
08-23 11:16:40.390 25484 25569 I TSLocationManager:     "priority": 0,
08-23 11:16:40.390 25484 25569 I TSLocationManager:     "sticky": false,
08-23 11:16:40.390 25484 25569 I TSLocationManager:     "strings": {},
08-23 11:16:40.390 25484 25569 I TSLocationManager:     "actions": []
08-23 11:16:40.390 25484 25569 I TSLocationManager:   },
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "params": {},
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "persist": true,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "persistMode": 2,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "schedule": [],
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "scheduleUseAlarmManager": false,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "speedJumpFilter": 300,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "startOnBoot": false,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "stationaryRadius": 10,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "stopAfterElapsedMinutes": 0,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "stopOnStationary": false,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "stopOnTerminate": true,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "stopTimeout": 5,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "triggerActivities": "in_vehicle, on_bicycle, on_foot, running, walking",
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "url": "",
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "useSignificantChangesOnly": false,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "enabled": false,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "schedulerEnabled": false,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "trackingMode": 1,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "odometer": 15.07443904876709,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "isFirstBoot": false,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "didLaunchInBackground": false,
08-23 11:16:40.390 25484 25569 I TSLocationManager:   "didDeviceReboot": false
08-23 11:16:40.390 25484 25569 I TSLocationManager: }
08-23 11:16:40.390 25484 25569 I TSLocationManager: [c.t.l.logger.LoggerFacade$a a] 
08-23 11:16:40.390 25484 25569 I TSLocationManager: ╔═════════════════════════════════════════════
08-23 11:16:40.390 25484 25569 I TSLocationManager: ║ DEVICE SENSORS
08-23 11:16:40.390 25484 25569 I TSLocationManager: ╠═════════════════════════════════════════════
08-23 11:16:40.390 25484 25569 I TSLocationManager: ╟─ ✅  ACCELEROMETER: {Sensor name="bmi220 Accelerometer Non-wakeup", vendor="BOSCH", version=16777503, type=1, maxRange=78.4532, resolution=0.0023928226, power=0.18, minDelay=5000}
08-23 11:16:40.390 25484 25569 I TSLocationManager: ╟─ ✅  GYROSCOPE: {Sensor name="bmi220 Gyroscope Non-wakeup", vendor="BOSCH", version=16777503, type=4, maxRange=17.452778, resolution=5.326165E-4, power=0.9, minDelay=5000}
08-23 11:16:40.390 25484 25569 I TSLocationManager: ╟─ ✅  MAGNETOMETER: {Sensor name="ak0991x Magnetometer Non-wakeup", vendor="akm", version=146966, type=2, maxRange=4912.0503, resolution=0.15, power=1.1, minDelay=10000}
08-23 11:16:40.390 25484 25569 I TSLocationManager: ╟─ ✅  SIGNIFICANT_MOTION: {Sensor name="sns_smd  Wakeup", vendor="qualcomm", version=1, type=17, maxRange=1.0, resolution=1.0, power=0.025, minDelay=-1}
08-23 11:16:40.390 25484 25569 I TSLocationManager: ╚═════════════════════════════════════════════
08-23 11:16:40.726 25484 25484 D TSLocationManager: [c.t.l.l.LifecycleManager onCreate] ☯️  onCreate
08-23 11:16:40.727 25484 25484 D TSLocationManager: [c.t.l.l.LifecycleManager onStart] ☯️  onStart
08-23 11:16:40.727 25484 25484 D TSLocationManager: [c.t.l.l.LifecycleManager onResume] ☯️  onResume
08-23 11:16:40.728 25484 25484 I TSLocationManager: [c.t.l.a.BackgroundGeolocation <init>] 
08-23 11:16:40.728 25484 25484 I TSLocationManager:   ✅  Google Play Services: connected (version code:12451000)
08-23 11:16:40.755 25484 25484 I TSLocationManager: [c.t.l.p.TSProviderManager startMonitoring] 
08-23 11:16:40.755 25484 25484 I TSLocationManager:   🎾  Start monitoring location-provider changes
08-23 11:16:40.769 25484 25569 D TSLocationManager: [c.t.l.data.sqlite.b prune] 
08-23 11:16:40.769 25484 25569 D TSLocationManager:   ℹ️  PRUNE -1 days
08-23 11:16:44.832 25484 25575 D TSLocationManager: [c.t.l.adapter.TSConfig d] ℹ️   Persist config, dirty: [backgroundPermissionRationale, backgroundPermissionRationale.title, backgroundPermissionRationale.message, backgroundPermissionRationale.positiveAction, backgroundPermissionRationale.negativeAction, desiredAccuracy, disableLocationAuthorizationAlert, distanceFilter, headlessJobService, isMoving, logLevel, notification, notification.text, stationaryRadius]
08-23 11:16:49.249 25484 25575 I TSLocationManager: [c.t.locationmanager.util.c g] 
08-23 11:16:49.249 25484 25575 I TSLocationManager:   🔵  LocationAuthorization: Requesting Background permission
08-23 11:16:49.253 25484 25575 I TSLocationManager: [c.t.locationmanager.util.c e] 
08-23 11:16:49.253 25484 25575 I TSLocationManager:   🔵  Should show backgroundPermissionRationale? true
08-23 11:16:49.537 25484 25484 D TSLocationManager: [c.t.l.a.TSLocationManagerActivity execute] android.permission.ACCESS_BACKGROUND_LOCATION
08-23 11:20:07.837 25484 25484 D TSLocationManager: [c.t.l.a.TSLocationManagerActivity stop] eventCount: 0
08-23 11:20:07.893 25484 25575 I TSLocationManager: [c.t.locationmanager.util.c g] 
08-23 11:20:07.893 25484 25575 I TSLocationManager:   🔵  LocationAuthorization: Requesting Background permission
08-23 11:20:07.903 25484 25575 I TSLocationManager: [c.t.locationmanager.util.c e] 
08-23 11:20:07.903 25484 25575 I TSLocationManager:   🔵  Should show backgroundPermissionRationale? true
08-23 11:20:07.953 25484 25484 D TSLocationManager: [c.t.l.a.TSLocationManagerActivity onDestroy] 
08-23 11:20:08.145 25484 25484 D TSLocationManager: [c.t.l.a.TSLocationManagerActivity execute] android.permission.ACCESS_BACKGROUND_LOCATION
08-23 11:20:09.716 25484 25484 D TSLocationManager: [c.t.l.a.TSLocationManagerActivity stop] eventCount: 0
08-23 11:20:09.751 25484 26657 I TSLocationManager: - Enable: false → true, trackingMode: 1
08-23 11:20:09.764 25484 26657 D TSLocationManager: [c.t.l.http.HttpService startMonitoringConnectivityChanges] 
08-23 11:20:09.764 25484 26657 D TSLocationManager:   🎾  Start monitoring connectivity changes
08-23 11:20:09.771 25484 26657 D TSLocationManager: [c.t.l.device.DeviceSettings startMonitoringPowerSaveChanges] 
08-23 11:20:09.771 25484 26657 D TSLocationManager:   🎾  Start monitoring powersave changes
08-23 11:20:09.775 25484 25580 D TSLocationManager: [c.t.l.http.HttpService a] 
08-23 11:20:09.775 25484 25580 D TSLocationManager: ╔═════════════════════════════════════════════
08-23 11:20:09.775 25484 25580 D TSLocationManager: ║ 📶  Connectivity change: connected? true
08-23 11:20:09.775 25484 25580 D TSLocationManager: ╠═════════════════════════════════════════════
```

</details>

**Comment by christocracy** [comment]
Show me a screenshot of what you *think* is the "Location Authorization Alert".

**Comment by giovanegomes** [comment]
Sorry, it is not the "Location Authorization Alert".

Actually it is happening with the `backgroundPermissionRationale` alert.
There is a way to manage when to show it?
We need to show it just when executting `BackgroundGeolocation.requestPermission()`

This is the alert:
![alert](https://github.com/transistorsoft/react-native-background-geolocation/assets/77764303/5e497c04-ac6e-407e-95f0-5bf39e2f5b15)

**Comment by giovanegomes** [comment]
Yes, I want to request `Always`. But with the following code it was requesting twice: 
```
const state = await BackgroundGeolocation.ready(GEOLOCATION_CONFIG);

if (state.enabled) return;

try {
    await BackgroundGeolocation.requestPermission();
} catch (error) {
    console.error('Request permission', error);
}

await this.requestPermissions();
await BackgroundGeolocation.start();
```
First when executing: `await BackgroundGeolocation.requestPermission();`
and then when executing: `await BackgroundGeolocation.start();`

That's why I was wondering if has an way to manage when to show the alert

But I found another solution without using `BackgroundGeolocation.requestPermission();`. So I'll close this issue.
Thank's

**Comment by thongquach** [comment]
What is your approach, @giovanegomes? I had the same problem.
I tried to`disableLocationAuthorizationAlert` and create a custom alert myself. Nonetheless, doing this causes the location permission to disappear from app settings on iOS :)

**Comment by christocracy** [comment]
Config.locationAuthorizationRequest represents the permission the plug-in *expects* to be provided.  When the user doesn’t provide that expectation, locationAuthorizationAlert is shown.

If you don’t care if the user selects Always or WhenInUse, use locationAuthorizationRequest: ‘Any’.  Read the api docs.

**Comment by thongquach** [comment]
Hi @christocracy , my app actually want `Always`. The problem is that I don't want to constantly show the locationAuthorizationAlert, only once a day. Is there anyway I can achieve this behavior?
Maybe changing it to `Any` when I don't want to show the alert, and `Always` when I want to show it?

**Comment by christocracy** [comment]
Why don’t you go experiment?

**Comment by giovanegomes** [comment]
Hi @thongquach. In my case I'm using the library `react-native-permissions` to manage some permissions. So I changed to validate the location permission with this library before start the plugin.

**Comment by thongquach** [comment]
The issue is that if you do not ask for location permission by `disableLocationAuthorizationAlert`,  then the location permission does not appear on iOS app settings. Mentioned here: https://github.com/transistorsoft/react-native-background-geolocation/issues/1791#issuecomment-1699524291
To reproduce, you need to delete the app completely, then reset privacy and location, then install the app: https://stackoverflow.com/a/65220590/21617288

**Comment by UnderTheMoonspell** [comment]
I would also like to just prevent the dialog to be shown on start, but to be able to show it when calling requestPermission later on. The disableLocationAuthorizationAlert seems to be too strict

**Comment by christocracy** [comment]
The plug-in requests its required permissions **when you execute the `.start()` method.**

You are not forced to call .start().  It is completely up to you.

**Comment by UnderTheMoonspell** [comment]
@christocracy im not calling .start again (i only call startGeofences() once on user setup and never again).  Im just calling .ready again (as the docs say to always call it and not hide it behind a button)

**Comment by christocracy** [comment]
> (i only call startGeofences()

Same thing.  When you call .start() or .startGeofences(), the plug-in **necessarily** requests permissions.

**Comment by UnderTheMoonspell** [comment]
Maybe im missing something but my flow is as follows. The app ALWAYS calls the .ready() method. In our signup flow the user will have a page explaining why we collect the location data and will have a button to confirm and accept. On click i call the .startGeofences() and then the permissions will be required.

When the user gets back to the app, the .ready() method will be called as usual, but now ill only .requestPermission() to see if the user has given "Always" permissions, and if it hasnt then show that locationAuthortizationAlert.

Do I have to call .startGeofences() everytime? Even if I do, then I will still have the same problem, as the locationAuthorizationAlert is being called on app load, and the only thing done on app load is the .ready() not the startGeofences()

**Comment by christocracy** [comment]
Show me screenshots and/or screen recording of what you're talking about.

When you call .start() or .startGeofences(), it permanently sets a flag State.enabled, which persists through app terminate / device reboot.  When you call .stop(), State.enabled is permanently set to false.

You do not have to "re-execute" .start() / .startGeofences() with each boot.  You only need to call .ready with each boot and the plug-in will automatically adjust its configuration according to State.enabled.

**Comment by UnderTheMoonspell** [comment]
Thats exactly what im doing. I only call .startGeofences once, but still the locationAuthorizationAlert shows up on boot. And I would like to prevent that, and only call it when I manually call .requestPermission()


Uploading Simulator Screen Recording - iPhone 15 - 2024-01-12 at 16.10.28.mp4…

**Comment by christocracy** [comment]
Show me a screenshot of what you *think* is the "location authorization alert".

**Comment by UnderTheMoonspell** [comment]
@christocracy the last one in the video

![image](https://github.com/transistorsoft/react-native-background-geolocation/assets/10156878/a8ca0ee3-fbe7-41ce-a470-d5e4c4b108fd)

**Comment by christocracy** [comment]
@UnderTheMoonspell I see that you have jumped into this *closed* issue.  You are not the original poster.

I suggest you post a new issue and provide your own config, just as the OP up top.

**Comment by UnderTheMoonspell** [comment]
@christocracy got it 👍
